daemon/containerd/image_builder.go

```diff
@@ -152,26 +152,25 @@ func (i *ImageService) pullForBuilder(ctx context.Context, name string, authConf
 
 	img, err := i.GetImage(ctx, name, backend.GetImageOpts{Platform: platform})
 	if err != nil {
-		if cerrdefs.IsNotFound(err) && img != nil && platform != nil {
-			imgPlat := ocispec.Platform{
-				OS:           img.OS,
-				Architecture: img.BaseImgArch(),
-				Variant:      img.BaseImgVariant(),
-			}
+		if !cerrdefs.IsNotFound(err) || img == nil || platform == nil {
+			return nil, err
```

Comment:
Interesting; this is existing, but looks like this code-path can return a `nil, nil` - we should check the call-sites to make sure that's properly handled, and not resulting in a panic.

Possibly this should _create_ a not-found error when failing to resolve.